### ISSUE TITLE
Deploying a version for a separate area 

### ISSUE THREAD

**Issue by glitchedmob** [code-for-chapel-hill/NC-COVID-Support#131]
I'm working with [Hack4Good](https://hack4goodsgf.com/) from Springfield MO and we've been looking for ways to help our community respond to COVID-19. This project seems like a really great way to do that. There's just a couple things I wanted to check on before deploying our own version.

- Do you have any issues with us changing the text and images in our version of the site to match our area? e.g. changing the favicon and updating references of Orange County?
- Is it possible for you to clone the spreadsheet(s) and Google Forms in a way that we can copy them? That way we don't have to spend a lot of time recreating everything.

**Comment by readingdancer** [comment]
Hi @glitchedmob 

No problem with you changing your version for your own local branding, clearly it wouldn't make sense to people otherwise :)

What would be great would be if you can do this in your own Fork, but still contribute changes to the core project so we can all benefit. Honestly not sure how this works in practice, but I am pretty sure it's possible. Or we can potentially look to refactor certain bits like that so that they are pulled out into config files that could be more easily separated, perhaps you and I could have a Zoom chat about it tomorrow?

Yes, it would be easy to clone the spreadsheet and forms, please keep in mind that I intend to build a "real" backend based around the Open Source Umbraco CMS ( which I have been using for 12+ years ) so this will change how the data is stored and edited.

We are reaching the point where an Google spreadsheet is becoming a bit unmanageable and would like the flexibility Umbraco will give us.

Feel free to ping me on Slack if you are on the CfA or Open NC Collaborative slack channels. 

Cheers,

Chris

**Comment by KM-Hanson** [comment]
Hi @glitchedmob - Yep - the project is intended to be replicated & shared. Hope it can help! I'll go in and add a spreadsheet template on GitHub tonight. You will see that the second tab has metadata that describes the intended data types (to integrate with the code).

Note that we do adapt the spreadsheet as needed to fit our data and to interface with the code. So we should find a way to coordinate if you want to keep in line with the core project. But you are welcome to adapt it to your needs and to send us your feedback.

I'd be happy to help you get started on the spreadsheet and the project- feel free to email us at contact@nccovidsupport.org.

Thanks!

**Comment by KM-Hanson** [comment]
OK - @glitchedmob I just added a link to the spreadsheet template on the ReadMe section (just below the link to the live page):

![Screen Shot 2020-04-16 at 10 54 15 PM](https://user-images.githubusercontent.com/58799043/79527384-73a47080-8035-11ea-9378-3fcf9548728c.png)

**Comment by readingdancer** [comment]
Hi @glitchedmob 

Just a quick update, I have actually moved the link that @KM-Hanson added earlier this evening to a new page called [Replicating the site](https://github.com/code-for-chapel-hill/NC-COVID-Support/blob/master/docs/replicating-the-site.md)

In summary, it has the Google Sheet template link and ways to contact me to chat about the technical implementation.

Cheers, Chris

**Comment by KM-Hanson** [comment]
@glitchedmob Can you tell that we are happy at the thought of being able to share :)

One more point - you are welcome to change to look of the site as you build your own, but you do not have to. We are happy to make our graphic design and social media elements available so that you can get up-and-running quickly. It might be nice to have some visual consistency as we build a network of sites that are all contributing to this same framework. 

I'll work on adding those elements to GitHub and details to the "Replicating the site" page that @readingdancer made (thanks!).

In the mean time, you can check out our [Instagram page](https://www.instagram.com/nccovidsupport/) for examples of social media templates that we could provide you. And email us at contact@nccovidsupport.org with any questions.

Thanks!

**Comment by jason-klein** [comment]
@readingdancer @KM-Hanson Thank you for your support. Just an update since @glitchedmob has been tied up this weekend..

A team here in Springfield MO has been working to put data together and are excited to get this up and running. Thanks for adding the Replicating the Site doc. You can view our progress at a temporary URL http://covidsupport.sgf.dev/

I believe our intent is to launch an identical version (minus customizations required for our Springfield-specific deploy) and will PR changes to your upstream code. Per your suggestions, it would be helpful if most/all location-specific details could be moved to a config file (e.g. Location Name "NC", Map Center, Google Sheets JSON URL, etc). This would allow us to easily continue using the upstream code without having to fork a copy and merge our local changes.

**Comment by glitchedmob** [comment]
Finally got a chance to get back to all this (got real tied up with work stuff the past few days).

@readingdancer a few notes on a real backend for this project.
- One of the things we really liked about this project was how little overhead there was in terms of infrastructure. Since it's all FE code, hosting our own version was not only easy, but also free using Netlify. 
- @jason-klein mentioned to me was how handy it was to have a spreadsheet to use when compiling data for the site. Being able to manipulate many records at once helped the process go quite a bit faster
- With all that being said, I still see a ton of value in a real backend. A spreadsheet can only go so far before issues come up with so much data being loaded into memory on every page load. When it comes to a backend technology I did want to say that although Umbraco is a great CMS (I haven't used it myself but know plenty who have and love it) it only being able to run on Windows is a real bummer. Anyone who wants to contribute any backend code would have to be running Windows and hosting is generally more expensive for Windows servers. Ultimately choose whatever technology you see fit for that functionality, just wanted to add my two cents. 

Also, thanks for adding that replicating the site document. Once we launch ours I'll do a diff to see everything that we changed and make a PR to that document that provides a checklist of all the things we had to change to launch our version. That way other users don't miss some of the little things that are easily overlooked like updating Google Analytics or leaving improper metadata in the index.html file.

At this point though, I think that you all have answered the main questions so I'll go ahead and close this issue.

**Comment by readingdancer** [comment]
Hi @glitchedmob 

I just thought I'd reply to your message in the same thread in case anyone else finds this discussion.

I am planning to re-factor so that the data ( and it's filtering ) can be swapped out, that way you could continue to use the online spreadsheet and the client side filtering, you could use the Umbraco implementation or implement your own. This also has the added benefit that we can continue with the current implementation adding features while in parallel work on the new back office implementation knowing it will be a clean swap.

These days I don't think hosting for Umbraco is prohibitively expensive, for these sites a small VM would probably be fine or using Umbraco Cloud as it's a really nice dev experience.

I agree about the easy editing in a spreadsheet for multiple lines, but I can see from our site that now we have 500+ lines, it's starting to be a less fun editing experience, this is where using Umbraco I could produce nice ways to filter the data when editing etc. We could also potentially enable data to be imported and exported as a CSV for bulk editing.

Not relevant for now, but Umbraco are currently working on their next version based on .NET Core, which I believe will eventually be cross platform :)

We have not chatted about how you were going to implement your branding etc to make it easy for you to continue to pull in updates, what have you decided to do on this at the moment?

Thank for the PR suggestion on the docs, that took me a good few hours to write, but I'm not surprised if I've missed out some bits :)

Cheers,

Chris

**Comment by readingdancer** [comment]
@glitchedmob @jason-klein

Another quick FYI, you might have noticed that I've started adding some unit tests using:

https://vue-test-utils.vuejs.org and https://jestjs.io/en/

I am hoping this will help us all in the long run to ensure things work smoothly across 2 or more sites based on the same code base. Feel free to add more tests :)

**Comment by readingdancer** [comment]
Final thought for the night, if there are things you need changed to enable the easy white labeling of the site, please do raise issues here on GitHub, that way I know what the pain points are and we can discuss how best to implement a fix to ensure it fits in with all the other changes :)
